File: scripts/grep_file.py

import sys

filename = sys.argv[1]
search_term = sys.argv[2]

try:
    with open(filename, 'r', encoding='utf-8') as f:
        lines = f.readlines()
        for i, line in enumerate(lines):
            if search_term.lower() in line.lower():
                print(f"{i+1}: {line.strip()}")
except Exception as e:
    print(f"Error: {e}")

File: scripts/grep_file.js

const fs = require('fs');
const filename = process.argv[2];
const searchTerm = process.argv[3];

try {
    const content = fs.readFileSync(filename, 'utf8');
    const lines = content.split('\n');
    lines.forEach((line, i) => {
        if (line.toLowerCase().includes(searchTerm.toLowerCase())) {
            console.log(`${i + 1}: ${line.trim()}`);
        }
    });
} catch (e) {
    console.error(e);
}
